In [123]:
import pandas as pd


# 5 Questions
- Predict if patient survives stroke or not.
- Predict if patient will have disability.
- What are the main factors contributing to death?
- How does our model's predicted probability of death at 14 days/6 months align with the doctors predictions?
- What are positive factors for stroke survival?

In [124]:
stroke_trials_df = pd.read_csv('../../Downloads/V2_International_Stroke_Trials.csv')
stroke_trials_df.head()

C:\Users\Tom\AppData\Local\Temp\ipykernel_51712\1795348831.py:1: DtypeWarning: Columns (25) have mixed types. Specify dtype option on import or set low_memory=False.
  stroke_trials_df = pd.read_csv('../../Downloads/V2_International_Stroke_Trials.csv')


SEX  AGE RSLEEP RATRIAL RCT RVISINF RHEP24  RSBP RCONSC RDEF1  ... DEAD8  \
0   M   69      Y     NaN   Y       Y    NaN   140      D     N  ...     0   
1   M   76      Y     NaN   Y       N    NaN   150      F     Y  ...     0   
2   F   71      N     NaN   Y       N    NaN   170      F     Y  ...     0   
3   M   81      N     NaN   N       N    NaN   170      F     N  ...     0   
4   M   78      N     NaN   N       N    NaN   170      F     Y  ...     0   

  H14 ISC14 NK14 STRK14 HTI14 PE14 DVT14 TRAN14 NCB14  
0   0     0    0      0     0    0     0      0     0  
1   0     0    0      0     0    0     0      0     0  
2   0     0    0      0     0    0     0      0     0  
3   0     0    0      0     0    0     0      0     0  
4   0     0    0      0     0    0     0      0     0  

[5 rows x 78 columns]

In [125]:
print(stroke_trials_df.columns)

Index(['SEX', 'AGE', 'RSLEEP', 'RATRIAL', 'RCT', 'RVISINF', 'RHEP24', 'RSBP',
       'RCONSC', 'RDEF1', 'RDEF2', 'RDEF3', 'RDEF4', 'RDEF5', 'RDEF6', 'RDEF7',
       'RDEF8', 'STYPE', 'RDATE', 'RXASP', 'RXHEP', 'DASP14', 'DASPLT',
       'DLH14', 'DMH14', 'DHH14', 'ONDRUG', 'DSCH', 'DIVH', 'DAP', 'DOAC',
       'DGORM', 'DSTER', 'DCAA', 'DHAEMD', 'DCAREND', 'DTHROMB', 'DMAJNCH',
       'DDIAGISC', 'DDIAGHA', 'DDIAGUN', 'DNOSTRK', 'DRSISC', 'DRSH', 'DRSUNK',
       'DPE', 'DALIVE', 'DPLACE', 'DDEAD', 'DDEADC', 'FPLACE', 'FAP', 'FOAC',
       'COUNTRY', 'CNTRYNUM', 'EXPDD', 'EXPD6', 'EXPD14', 'SET14D', 'ID14',
       'OCCODE', 'DEAD1', 'DEAD2', 'DEAD3', 'DEAD4', 'DEAD5', 'DEAD6', 'DEAD7',
       'DEAD8', 'H14', 'ISC14', 'NK14', 'STRK14', 'HTI14', 'PE14', 'DVT14',
       'TRAN14', 'NCB14'],
      dtype='object')


In [126]:
stroke_trials_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   19435  19435.000000  19435   18451  19435   19435  19091   
unique      2           NaN      2       2      2       2      2   
top         M           NaN      N       N      Y       N      N   
freq    10407           NaN  13750   15282  13024   13020  18655   
mean      NaN     71.715410    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.619714    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     99.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ...         DEAD8           H14  \
count   19435.000000  19435  19435  ...  19435.000000  19435.000000   
unique           NaN      3      3  ...           NaN           NaN   
top              NaN      F      Y  ...           NaN           NaN   
freq             NaN  14921  14099  ...           NaN           NaN   
mean      160.159197    NaN    NaN  ...      0.011834      0.008284   
std        27.610382    NaN    NaN  ...      0.108143      0.090641   
min        70.000000    NaN    NaN  ...      0.000000      0.000000   
25%       140.000000    NaN    NaN  ...      0.000000      0.000000   
50%       160.000000    NaN    NaN  ...      0.000000      0.000000   
75%       180.000000    NaN    NaN  ...      0.000000      0.000000   
max       295.000000    NaN    NaN  ...      1.000000      1.000000   

               ISC14          NK14        STRK14         HTI14          PE14  \
count   19435.000000  19435.000000  19435.000000  19435.000000  19435.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.020324      0.013378      0.041523      0.003036      0.006895   
std         0.141110      0.114890      0.199502      0.055015      0.082750   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.000000      0.000000      0.000000      0.000000      0.000000   
50%         0.000000      0.000000      0.000000      0.000000      0.000000   
75%         0.000000      0.000000      0.000000      0.000000      0.000000   
max         1.000000      1.000000      1.000000      1.000000      1.000000   

               DVT14        TRAN14         NCB14  
count   19435.000000  19435.000000  19435.000000  
unique           NaN           NaN           NaN  
top              NaN           NaN           NaN  
freq             NaN           NaN           NaN  
mean        0.001132      0.008541      0.027322  
std         0.033627      0.092026      0.163024  
min         0.000000      0.000000      0.000000  
25%         0.000000      0.000000      0.000000  
50%         0.000000      0.000000      0.000000  
75%         0.000000      0.000000      0.000000  
max         1.000000      1.000000      1.000000  

[11 rows x 78 columns]

In [127]:
# Find the number of na values in each column
for col in stroke_trials_df.columns:
    if stroke_trials_df[col].isna().sum() > 0:
        print(f'{col}: {stroke_trials_df[col].isna().sum()}')

RATRIAL: 984
RHEP24: 344
DASP14: 22
DASPLT: 147
DLH14: 22
DMH14: 1006
DHH14: 18451
ONDRUG: 1
DSCH: 305
DIVH: 305
DAP: 18
DOAC: 18
DGORM: 23
DSTER: 28
DCAA: 29
DHAEMD: 28
DCAREND: 1005
DTHROMB: 315
DMAJNCH: 14
DDIAGISC: 23
DDIAGHA: 26
DDIAGUN: 23
DNOSTRK: 26
DRSISC: 18
DRSH: 15
DRSUNK: 1007
DPE: 14
DALIVE: 28
DPLACE: 9709
DDEAD: 19
DDEADC: 17101
FPLACE: 4502
FAP: 4568
FOAC: 4574


In [128]:
# Fill na values in DDEADC and FDEADC column with 0 and in DRSUNK, DPLACE, FPLACE, FAP, FOAC columns with 'U'
stroke_trials_df['DDEADC'] = stroke_trials_df['DDEADC'].fillna(0)
stroke_trials_df['DASP14'] = stroke_trials_df['DASP14'].fillna('U')
stroke_trials_df['DASPLT'] = stroke_trials_df['DASPLT'].fillna('U')
stroke_trials_df['DRSUNK'] = stroke_trials_df['DRSUNK'].fillna('U')
stroke_trials_df['DPLACE'] = stroke_trials_df['DPLACE'].fillna('U')
stroke_trials_df['FPLACE'] = stroke_trials_df['FPLACE'].fillna('U')
stroke_trials_df['DMH14'] = stroke_trials_df['DMH14'].fillna('U')
stroke_trials_df['DHH14'] = stroke_trials_df['DHH14'].fillna('U')
stroke_trials_df['DLH14'] = stroke_trials_df['DLH14'].fillna('U')
stroke_trials_df['DSCH'] = stroke_trials_df['DSCH'].fillna('U')
stroke_trials_df['DIVH'] = stroke_trials_df['DIVH'].fillna('U')
stroke_trials_df['DCAREND'] = stroke_trials_df['DCAREND'].fillna('U')
stroke_trials_df['DTHROMB'] = stroke_trials_df['DTHROMB'].fillna('U')
stroke_trials_df['DALIVE'] = stroke_trials_df['DALIVE'].fillna('U')
stroke_trials_df['FAP'] = stroke_trials_df['FAP'].fillna('U')
stroke_trials_df['FOAC'] = stroke_trials_df['FOAC'].fillna('U')
stroke_trials_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   19435  19435.000000  19435   18451  19435   19435  19091   
unique      2           NaN      2       2      2       2      2   
top         M           NaN      N       N      Y       N      N   
freq    10407           NaN  13750   15282  13024   13020  18655   
mean      NaN     71.715410    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.619714    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     99.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ...         DEAD8           H14  \
count   19435.000000  19435  19435  ...  19435.000000  19435.000000   
unique           NaN      3      3  ...           NaN           NaN   
top              NaN      F      Y  ...           NaN           NaN   
freq             NaN  14921  14099  ...           NaN           NaN   
mean      160.159197    NaN    NaN  ...      0.011834      0.008284   
std        27.610382    NaN    NaN  ...      0.108143      0.090641   
min        70.000000    NaN    NaN  ...      0.000000      0.000000   
25%       140.000000    NaN    NaN  ...      0.000000      0.000000   
50%       160.000000    NaN    NaN  ...      0.000000      0.000000   
75%       180.000000    NaN    NaN  ...      0.000000      0.000000   
max       295.000000    NaN    NaN  ...      1.000000      1.000000   

               ISC14          NK14        STRK14         HTI14          PE14  \
count   19435.000000  19435.000000  19435.000000  19435.000000  19435.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.020324      0.013378      0.041523      0.003036      0.006895   
std         0.141110      0.114890      0.199502      0.055015      0.082750   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.000000      0.000000      0.000000      0.000000      0.000000   
50%         0.000000      0.000000      0.000000      0.000000      0.000000   
75%         0.000000      0.000000      0.000000      0.000000      0.000000   
max         1.000000      1.000000      1.000000      1.000000      1.000000   

               DVT14        TRAN14         NCB14  
count   19435.000000  19435.000000  19435.000000  
unique           NaN           NaN           NaN  
top              NaN           NaN           NaN  
freq             NaN           NaN           NaN  
mean        0.001132      0.008541      0.027322  
std         0.033627      0.092026      0.163024  
min         0.000000      0.000000      0.000000  
25%         0.000000      0.000000      0.000000  
50%         0.000000      0.000000      0.000000  
75%         0.000000      0.000000      0.000000  
max         1.000000      1.000000      1.000000  

[11 rows x 78 columns]

In [129]:
# Find the number of na values in each column
for col in stroke_trials_df.columns:
    if stroke_trials_df[col].isna().sum() > 0:
        print(f'{col}: {stroke_trials_df[col].isna().sum()}')

RATRIAL: 984
RHEP24: 344
ONDRUG: 1
DAP: 18
DOAC: 18
DGORM: 23
DSTER: 28
DCAA: 29
DHAEMD: 28
DMAJNCH: 14
DDIAGISC: 23
DDIAGHA: 26
DDIAGUN: 23
DNOSTRK: 26
DRSISC: 18
DRSH: 15
DPE: 14
DDEAD: 19


In [130]:
# View the rows that have NaN values in RHEP24 column and see how many have 'N', 'U', or NaN values in the other heparin columns (RXHEP, DLH14, DHH14, DMH14, DSCH, DIVH)
rhep24 = stroke_trials_df.loc[stroke_trials_df['RHEP24'].isnull()]
rhep24[['RXHEP','DLH14','DHH14','DMH14','DSCH','DIVH']].value_counts()

RXHEP  DLH14  DHH14  DMH14  DSCH  DIVH
N      N      N      U      U     U       147
L      Y      N      U      U     U        61
H      N      Y      U      U     U        58
N      N      N      U      N     N        23
L      Y      N      U      N     N        13
H      N      N      U      U     U        11
              Y      U      N     N        10
L      N      N      U      U     U         7
H      N      N      U      N     N         5
       Y      N      U      N     N         2
L      N      Y      U      N     N         1
       U      N      U      U     U         1
       N      N      U      N     U         1
                                  N         1
       Y      N      U      Y     N         1
H      Y      N      U      U     U         1
N      N      N      U      Y     Y         1
Name: count, dtype: int64

There are 173 rows that we can use if we save the NaN rows in the 'RHEP24' column (147 + 23 + 1 + 1 + 1)

In [131]:
# Fill na values in RHEP24 column with 'U'
stroke_trials_df['RHEP24'] = stroke_trials_df['RHEP24'].fillna('U')

In [132]:
# Drop all rows with na values
stroke_trials_dropna_df = stroke_trials_df.dropna()
stroke_trials_dropna_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   18405  18405.000000  18405   18405  18405   18405  18405   
unique      2           NaN      2       2      2       2      2   
top         M           NaN      N       N      Y       N      N   
freq     9825           NaN  13011   15243  12372   12382  17977   
mean      NaN     71.802934    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.609051    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ...         DEAD8           H14  \
count   18405.000000  18405  18405  ...  18405.000000  18405.000000   
unique           NaN      3      3  ...           NaN           NaN   
top              NaN      F      Y  ...           NaN           NaN   
freq             NaN  14112  13363  ...           NaN           NaN   
mean      160.213583    NaN    NaN  ...      0.011790      0.007987   
std        27.724123    NaN    NaN  ...      0.107944      0.089015   
min        70.000000    NaN    NaN  ...      0.000000      0.000000   
25%       140.000000    NaN    NaN  ...      0.000000      0.000000   
50%       160.000000    NaN    NaN  ...      0.000000      0.000000   
75%       180.000000    NaN    NaN  ...      0.000000      0.000000   
max       295.000000    NaN    NaN  ...      1.000000      1.000000   

               ISC14          NK14        STRK14         HTI14          PE14  \
count   18405.000000  18405.000000  18405.000000  18405.000000  18405.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.019886      0.013909      0.041293      0.002988      0.006792   
std         0.139612      0.117118      0.198973      0.054585      0.082133   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.000000      0.000000      0.000000      0.000000      0.000000   
50%         0.000000      0.000000      0.000000      0.000000      0.000000   
75%         0.000000      0.000000      0.000000      0.000000      0.000000   
max         1.000000      1.000000      1.000000      1.000000      1.000000   

               DVT14        TRAN14         NCB14  
count   18405.000000  18405.000000  18405.000000  
unique           NaN           NaN           NaN  
top              NaN           NaN           NaN  
freq             NaN           NaN           NaN  
mean        0.001087      0.008802      0.027330  
std         0.032948      0.093407      0.163046  
min         0.000000      0.000000      0.000000  
25%         0.000000      0.000000      0.000000  
50%         0.000000      0.000000      0.000000  
75%         0.000000      0.000000      0.000000  
max         1.000000      1.000000      1.000000  

[11 rows x 78 columns]

In [133]:
# Filter out rows that have 'Y' in the RHEP24 column
stroke_trials_rhep24_df = stroke_trials_dropna_df[stroke_trials_dropna_df['RHEP24'] != 'Y']
stroke_trials_rhep24_df = stroke_trials_rhep24_df.reset_index(drop=True)
stroke_trials_rhep24_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   17977  17977.000000  17977   17977  17977   17977  17977   
unique      2           NaN      2       2      2       2      1   
top         M           NaN      N       N      Y       N      N   
freq     9627           NaN  12687   14912  12039   12115  17977   
mean      NaN     71.787006    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.623209    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ...         DEAD8           H14  \
count   17977.000000  17977  17977  ...  17977.000000  17977.000000   
unique           NaN      3      3  ...           NaN           NaN   
top              NaN      F      Y  ...           NaN           NaN   
freq             NaN  13784  13034  ...           NaN           NaN   
mean      160.310174    NaN    NaN  ...      0.011960      0.007621   
std        27.749877    NaN    NaN  ...      0.108708      0.086967   
min        70.000000    NaN    NaN  ...      0.000000      0.000000   
25%       140.000000    NaN    NaN  ...      0.000000      0.000000   
50%       160.000000    NaN    NaN  ...      0.000000      0.000000   
75%       180.000000    NaN    NaN  ...      0.000000      0.000000   
max       295.000000    NaN    NaN  ...      1.000000      1.000000   

               ISC14          NK14        STRK14         HTI14          PE14  \
count   17977.000000  17977.000000  17977.000000  17977.000000  17977.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.019859      0.013851      0.040830      0.002614      0.006675   
std         0.139519      0.116876      0.197902      0.051066      0.081431   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.000000      0.000000      0.000000      0.000000      0.000000   
50%         0.000000      0.000000      0.000000      0.000000      0.000000   
75%         0.000000      0.000000      0.000000      0.000000      0.000000   
max         1.000000      1.000000      1.000000      1.000000      1.000000   

               DVT14        TRAN14         NCB14  
count   17977.000000  17977.000000  17977.000000  
unique           NaN           NaN           NaN  
top              NaN           NaN           NaN  
freq             NaN           NaN           NaN  
mean        0.001113      0.008789      0.027257  
std         0.033337      0.093339      0.162836  
min         0.000000      0.000000      0.000000  
25%         0.000000      0.000000      0.000000  
50%         0.000000      0.000000      0.000000  
75%         0.000000      0.000000      0.000000  
max         1.000000      1.000000      1.000000  

[11 rows x 78 columns]

In [134]:
# Filter out rows that have 'Y' in the RXHEP column
stroke_trials_rxhep_df = stroke_trials_rhep24_df[stroke_trials_rhep24_df['RXHEP'] != 'Y']
stroke_trials_rxhep_df = stroke_trials_rxhep_df.drop(columns=['RXHEP'])
stroke_trials_rxhep_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   17977  17977.000000  17977   17977  17977   17977  17977   
unique      2           NaN      2       2      2       2      1   
top         M           NaN      N       N      Y       N      N   
freq     9627           NaN  12687   14912  12039   12115  17977   
mean      NaN     71.787006    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.623209    NaN     NaN    NaN     NaN    NaN   
min       NaN     16.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ...         DEAD8           H14  \
count   17977.000000  17977  17977  ...  17977.000000  17977.000000   
unique           NaN      3      3  ...           NaN           NaN   
top              NaN      F      Y  ...           NaN           NaN   
freq             NaN  13784  13034  ...           NaN           NaN   
mean      160.310174    NaN    NaN  ...      0.011960      0.007621   
std        27.749877    NaN    NaN  ...      0.108708      0.086967   
min        70.000000    NaN    NaN  ...      0.000000      0.000000   
25%       140.000000    NaN    NaN  ...      0.000000      0.000000   
50%       160.000000    NaN    NaN  ...      0.000000      0.000000   
75%       180.000000    NaN    NaN  ...      0.000000      0.000000   
max       295.000000    NaN    NaN  ...      1.000000      1.000000   

               ISC14          NK14        STRK14         HTI14          PE14  \
count   17977.000000  17977.000000  17977.000000  17977.000000  17977.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.019859      0.013851      0.040830      0.002614      0.006675   
std         0.139519      0.116876      0.197902      0.051066      0.081431   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.000000      0.000000      0.000000      0.000000      0.000000   
50%         0.000000      0.000000      0.000000      0.000000      0.000000   
75%         0.000000      0.000000      0.000000      0.000000      0.000000   
max         1.000000      1.000000      1.000000      1.000000      1.000000   

               DVT14        TRAN14         NCB14  
count   17977.000000  17977.000000  17977.000000  
unique           NaN           NaN           NaN  
top              NaN           NaN           NaN  
freq             NaN           NaN           NaN  
mean        0.001113      0.008789      0.027257  
std         0.033337      0.093339      0.162836  
min         0.000000      0.000000      0.000000  
25%         0.000000      0.000000      0.000000  
50%         0.000000      0.000000      0.000000  
75%         0.000000      0.000000      0.000000  
max         1.000000      1.000000      1.000000  

[11 rows x 77 columns]

In [135]:
# Filter out rows that have 'Y' in the DLH14 column
stroke_trials_dlh14_df = stroke_trials_rxhep_df[stroke_trials_rxhep_df['DLH14'] != 'Y']
stroke_trials_dlh14_df = stroke_trials_dlh14_df.drop(columns=['DLH14'])
stroke_trials_dlh14_df.describe(include='all')

SEX           AGE RSLEEP RATRIAL    RCT RVISINF RHEP24  \
count   13768  13768.000000  13768   13768  13768   13768  13768   
unique      2           NaN      2       2      2       2      1   
top         M           NaN      N       N      Y       N      N   
freq     7360           NaN   9718   11397   9151    9259  13768   
mean      NaN     71.765035    NaN     NaN    NaN     NaN    NaN   
std       NaN     11.644272    NaN     NaN    NaN     NaN    NaN   
min       NaN     19.000000    NaN     NaN    NaN     NaN    NaN   
25%       NaN     65.000000    NaN     NaN    NaN     NaN    NaN   
50%       NaN     73.000000    NaN     NaN    NaN     NaN    NaN   
75%       NaN     80.000000    NaN     NaN    NaN     NaN    NaN   
max       NaN     98.000000    NaN     NaN    NaN     NaN    NaN   

                RSBP RCONSC  RDEF1  ...         DEAD8           H14  \
count   13768.000000  13768  13768  ...  13768.000000  13768.000000   
unique           NaN      3      3  ...           NaN           NaN   
top              NaN      F      Y  ...           NaN           NaN   
freq             NaN  10552   9974  ...           NaN           NaN   
mean      160.071252    NaN    NaN  ...      0.012057      0.009006   
std        27.720367    NaN    NaN  ...      0.109144      0.094477   
min        71.000000    NaN    NaN  ...      0.000000      0.000000   
25%       140.000000    NaN    NaN  ...      0.000000      0.000000   
50%       160.000000    NaN    NaN  ...      0.000000      0.000000   
75%       180.000000    NaN    NaN  ...      0.000000      0.000000   
max       295.000000    NaN    NaN  ...      1.000000      1.000000   

               ISC14          NK14        STRK14         HTI14          PE14  \
count   13768.000000  13768.000000  13768.000000  13768.000000  13768.000000   
unique           NaN           NaN           NaN           NaN           NaN   
top              NaN           NaN           NaN           NaN           NaN   
freq             NaN           NaN           NaN           NaN           NaN   
mean        0.021862      0.014744      0.045032      0.003051      0.006682   
std         0.146239      0.120532      0.207382      0.055149      0.081474   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         0.000000      0.000000      0.000000      0.000000      0.000000   
50%         0.000000      0.000000      0.000000      0.000000      0.000000   
75%         0.000000      0.000000      0.000000      0.000000      0.000000   
max         1.000000      1.000000      1.000000      1.000000      1.000000   

               DVT14        TRAN14         NCB14  
count   13768.000000  13768.000000  13768.000000  
unique           NaN           NaN           NaN  
top              NaN           NaN           NaN  
freq             NaN           NaN           NaN  
mean        0.001235      0.009805      0.030651  
std         0.035119      0.098539      0.172376  
min         0.000000      0.000000      0.000000  
25%         0.000000      0.000000      0.000000  
50%         0.000000      0.000000      0.000000  
75%         0.000000      0.000000      0.000000  
max         1.000000      1.000000      1.000000  

[11 rows x 76 columns]

In [136]:
# Filer out rows that have 'Y' in the DMH14 column
stroke_trials_dmh14_df = stroke_trials_dlh14_df[stroke_trials_dlh14_df['DMH14'] != 'Y']
stroke_trials_dmh14_df = stroke_trials_dmh14_df.drop(columns=['DMH14'])
stroke_trials_dmh14_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9783  9783.000000   9783    9783  9783    9783   9783  9783.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    5203          NaN   6915    8094  6386    6658   9783          NaN   
mean     NaN    71.780435    NaN     NaN   NaN     NaN    NaN   160.170807   
std      NaN    11.709731    NaN     NaN   NaN     NaN    NaN    28.000386   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    80.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    9783  9783  ...  9783.000000  9783.000000  9783.000000  9783.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     7490  7076  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.012266     0.007871     0.022795     0.016355   
std       NaN   NaN  ...     0.110077     0.088372     0.149256     0.126843   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   9783.000000  9783.000000  9783.000000  9783.000000  9783.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.046714     0.002658     0.007769     0.001635     0.008382   
std        0.211036     0.051487     0.087801     0.040410     0.091173   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

              NCB14  
count   9783.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean       0.024328  
std        0.154073  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[11 rows x 75 columns]

In [137]:
# Filter out rows that have 'Y' in the DHH14 column
stroke_trials_dhh14_df = stroke_trials_dmh14_df[stroke_trials_dmh14_df['DHH14'] != 'Y']
stroke_trials_dhh14_df = stroke_trials_dhh14_df.drop(columns=['DHH14'])
stroke_trials_dhh14_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9783  9783.000000   9783    9783  9783    9783   9783  9783.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    5203          NaN   6915    8094  6386    6658   9783          NaN   
mean     NaN    71.780435    NaN     NaN   NaN     NaN    NaN   160.170807   
std      NaN    11.709731    NaN     NaN   NaN     NaN    NaN    28.000386   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    80.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    9783  9783  ...  9783.000000  9783.000000  9783.000000  9783.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     7490  7076  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.012266     0.007871     0.022795     0.016355   
std       NaN   NaN  ...     0.110077     0.088372     0.149256     0.126843   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   9783.000000  9783.000000  9783.000000  9783.000000  9783.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.046714     0.002658     0.007769     0.001635     0.008382   
std        0.211036     0.051487     0.087801     0.040410     0.091173   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

              NCB14  
count   9783.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean       0.024328  
std        0.154073  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[11 rows x 74 columns]

In [138]:
# Filter out rows that have 'Y' in the DSCH column
stroke_trials_dsch_df = stroke_trials_dhh14_df[stroke_trials_dhh14_df['DSCH'] != 'Y']
stroke_trials_dsch_df = stroke_trials_dsch_df.drop(columns=['DSCH'])
stroke_trials_dsch_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9446  9446.000000   9446    9446  9446    9446   9446  9446.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    5038          NaN   6669    7823  6139    6433   9446          NaN   
mean     NaN    71.780224    NaN     NaN   NaN     NaN    NaN   160.222316   
std      NaN    11.719854    NaN     NaN   NaN     NaN    NaN    28.011635   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    80.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    9446  9446  ...  9446.000000  9446.000000  9446.000000  9446.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     7260  6805  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.011751     0.007728     0.020008     0.016409   
std       NaN   NaN  ...     0.107769     0.087574     0.140036     0.127049   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   9446.000000  9446.000000  9446.000000  9446.000000  9446.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.043934     0.002647     0.007093     0.001165     0.008363   
std        0.204959     0.051380     0.083925     0.034107     0.091073   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

              NCB14  
count   9446.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean       0.024455  
std        0.154465  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[11 rows x 73 columns]

In [139]:
# Filter out rows that have 'Y' in the DIVH column
stroke_trials_divh_df = stroke_trials_dsch_df[stroke_trials_dsch_df['DIVH'] != 'Y']
stroke_trials_divh_df = stroke_trials_divh_df.drop(columns=['DIVH'])
stroke_trials_divh_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   9198  9198.000000   9198    9198  9198    9198   9198  9198.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4893          NaN   6486    7620  5938    6282   9198          NaN   
mean     NaN    71.887041    NaN     NaN   NaN     NaN    NaN   160.256143   
std      NaN    11.659757    NaN     NaN   NaN     NaN    NaN    28.077996   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    9198  9198  ...  9198.000000  9198.000000  9198.000000  9198.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     7057  6623  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.011850     0.007937     0.015438     0.015873   
std       NaN   NaN  ...     0.108218     0.088738     0.123294     0.124991   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   9198.000000  9198.000000  9198.000000  9198.000000  9198.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.039030     0.002718     0.005110     0.000652     0.008589   
std        0.193677     0.052066     0.071304     0.025534     0.092282   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

              NCB14  
count   9198.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean       0.024897  
std        0.155819  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[11 rows x 72 columns]

In [141]:
# Filter out rows that have 'Y' in the DHAEMD column
stroke_trials_dhaemd_df = stroke_trials_divh_df[stroke_trials_divh_df['DHAEMD'] != 'Y']
stroke_trials_dhaemd_df = stroke_trials_dhaemd_df.drop(columns=['DHAEMD'])
stroke_trials_dhaemd_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8925  8925.000000   8925    8925  8925    8925   8925  8925.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4750          NaN   6312    7392  5717    6097   8925          NaN   
mean     NaN    71.967619    NaN     NaN   NaN     NaN    NaN   160.238319   
std      NaN    11.641860    NaN     NaN   NaN     NaN    NaN    28.090209   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    8925  8925  ...  8925.000000  8925.000000  8925.000000  8925.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     6863  6401  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.011877     0.007843     0.015462     0.016134   
std       NaN   NaN  ...     0.108337     0.088218     0.123389     0.126000   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   8925.000000  8925.000000  8925.000000  8925.000000  8925.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.039216     0.002801     0.005154     0.000672     0.008739   
std        0.194119     0.052854     0.071611     0.025921     0.093081   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

              NCB14  
count   8925.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean       0.024874  
std        0.155750  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[11 rows x 71 columns]

In [142]:
# Filter out rows that have 'Y' in the DCAREND column
stroke_trials_dcarend_df = stroke_trials_dhaemd_df[stroke_trials_dhaemd_df['DCAREND'] != 'Y']
stroke_trials_dcarend_df = stroke_trials_dcarend_df.drop(columns=['DCAREND'])
stroke_trials_dcarend_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8912  8912.000000   8912    8912  8912    8912   8912  8912.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4741          NaN   6301    7379  5708    6088   8912          NaN   
mean     NaN    71.971836    NaN     NaN   NaN     NaN    NaN   160.224192   
std      NaN    11.640598    NaN     NaN   NaN     NaN    NaN    28.102942   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    8912  8912  ...  8912.000000  8912.000000  8912.000000  8912.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     6850  6394  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.011894     0.007855     0.015036     0.016158   
std       NaN   NaN  ...     0.108416     0.088282     0.121702     0.126090   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   8912.000000  8912.000000  8912.000000  8912.000000  8912.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.038824     0.002805     0.005162     0.000673     0.008752   
std        0.193186     0.052893     0.071663     0.025940     0.093148   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

             NCB14  
count   8912.00000  
unique         NaN  
top            NaN  
freq           NaN  
mean       0.02491  
std        0.15586  
min        0.00000  
25%        0.00000  
50%        0.00000  
75%        0.00000  
max        1.00000  

[11 rows x 70 columns]

In [143]:
# Filter out rows that have 'Y' in the DTHROMB column
stroke_trials_dthromb_df = stroke_trials_dcarend_df[stroke_trials_dcarend_df['DTHROMB'] != 'Y']
stroke_trials_dthromb_df = stroke_trials_dthromb_df.drop(columns=['DTHROMB'])
stroke_trials_dthromb_df.describe(include='all')

SEX          AGE RSLEEP RATRIAL   RCT RVISINF RHEP24         RSBP  \
count   8907  8907.000000   8907    8907  8907    8907   8907  8907.000000   
unique     2          NaN      2       2     2       2      1          NaN   
top        M          NaN      N       N     Y       N      N          NaN   
freq    4740          NaN   6297    7377  5704    6084   8907          NaN   
mean     NaN    71.970809    NaN     NaN   NaN     NaN    NaN   160.227686   
std      NaN    11.642180    NaN     NaN   NaN     NaN    NaN    28.108205   
min      NaN    19.000000    NaN     NaN   NaN     NaN    NaN    71.000000   
25%      NaN    65.000000    NaN     NaN   NaN     NaN    NaN   140.000000   
50%      NaN    74.000000    NaN     NaN   NaN     NaN    NaN   160.000000   
75%      NaN    81.000000    NaN     NaN   NaN     NaN    NaN   180.000000   
max      NaN    98.000000    NaN     NaN   NaN     NaN    NaN   295.000000   

       RCONSC RDEF1  ...        DEAD8          H14        ISC14         NK14  \
count    8907  8907  ...  8907.000000  8907.000000  8907.000000  8907.000000   
unique      3     3  ...          NaN          NaN          NaN          NaN   
top         F     Y  ...          NaN          NaN          NaN          NaN   
freq     6846  6390  ...          NaN          NaN          NaN          NaN   
mean      NaN   NaN  ...     0.011901     0.007859     0.015044     0.016167   
std       NaN   NaN  ...     0.108446     0.088307     0.121736     0.126125   
min       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
25%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
50%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
75%       NaN   NaN  ...     0.000000     0.000000     0.000000     0.000000   
max       NaN   NaN  ...     1.000000     1.000000     1.000000     1.000000   

             STRK14        HTI14         PE14        DVT14       TRAN14  \
count   8907.000000  8907.000000  8907.000000  8907.000000  8907.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       0.038846     0.002807     0.005164     0.000674     0.008757   
std        0.193238     0.052908     0.071683     0.025947     0.093174   
min        0.000000     0.000000     0.000000     0.000000     0.000000   
25%        0.000000     0.000000     0.000000     0.000000     0.000000   
50%        0.000000     0.000000     0.000000     0.000000     0.000000   
75%        0.000000     0.000000     0.000000     0.000000     0.000000   
max        1.000000     1.000000     1.000000     1.000000     1.000000   

              NCB14  
count   8907.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean       0.024924  
std        0.155903  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[11 rows x 69 columns]

In [ ]:
target_14_days_df = stroke_trials_dthromb_df[['DPLACE', 'DDEAD', 'DDEADC']]
target_6_months_df = stroke_trials_dthromb_df[['FPLACE', 'OCCODE', 'FDEADC']]

In [140]:
stroke_trials_dropna_df[['EXPD14', 'SET14D', 'ID14']].head(29)

EXPD14  SET14D  ID14
984   0.0736       1     0
985   0.0323       1     1
986   0.0375       1     0
987   0.0230       1     0
988   0.0162       1     0
989   0.2623       1     0
990   0.0885       1     0
991   0.0904       1     0
992   0.0543       1     0
993   0.0191       1     0
994   0.0368       1     0
996   0.0353       1     0
997   0.0340       1     0
998   0.0171       1     0
999   0.2715       1     0
1000  0.0791       1     0
1001  0.0316       1     0
1002  0.0199       1     0
1003  0.0603       1     0
1004  0.2823       1     0
1005  0.3068       1     0
1006  0.2831       1     0
1007  0.1311       1     0
1008  0.0228       1     0
1009  0.3799       1     0
1010  0.0260       1     0
1012  0.0350       1     0
1013  0.0811       1     0
1014  0.0557       1     0